sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggUtils.scala

```diff
@@ -27,26 +27,87 @@ import org.apache.spark.sql.execution.streaming.{StateStoreRestoreExec, StateSto
  */
 object AggUtils {
 
-  def planAggregateWithoutPartial(
+  private[execution] def isAggregate(operator: SparkPlan): Boolean = {
+    operator.isInstanceOf[HashAggregateExec] || operator.isInstanceOf[SortAggregateExec]
+  }
+
+  private[execution] def supportPartialAggregate(operator: SparkPlan): Boolean = {
```

Comment:
We could just merge this with `isAggregate(...)`. You could also turn this into an Extractor, if that makes life easier in `EnsureRequirements`.